src/libos/src/entry.rs

```diff
@@ -87,6 +89,28 @@ pub extern "C" fn occlum_ecall_init(log_level: *const c_char, instance_dir: *con
         unsafe { backtrace::enable_backtrace(&ENCLAVE_PATH, PrintFormat::Short) };
     });
 
+    let resolv_conf_bytes = unsafe {
```

Comment:
Can we use from_bytes_with_nul to avoid using unsafe function?

Reply:
I can try it

Reply:
Since the parsed parameter is a pointer, we can't use from_bytes_with_nul. The from_ptr method is unsafe for a number of reasons:

> There is no guarantee to the validity of ptr.
> The returned lifetime is not guaranteed to be the actual lifetime of ptr.
> There is no guarantee that the memory pointed to by ptr contains a valid nul terminator byte at the end of the string.
> It is not guaranteed that the memory pointed by ptr won’t change before the CStr has been destroyed.